### PR TITLE
allow to manage devices (unregister) when reaching max nodes

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
allow to manage devices (unregister) when reaching max nodes

# Impacts
Captive portal

# Delete branch after merge
YES

# NEWS file entries
## New Features
* Allow to manage devices (unregister) when reaching max nodes

### CODE REVIEW

**Comment by julsemaan** [comment]
@fdurand, if you have time to review this for 11.2, lets do it, otherwise we bump to 11.3

File: html/pfappserver/root/src/views/Configuration/connectionProfiles/_components/TheForm.vue [hunk 1]
```diff
@@ -138,6 +138,11 @@
           :text="$i18n.t('Under most circumstances we can redirect the user to the URL he originally intended to visit. However, you may prefer to force the captive portal to redirect the user to the redirection URL.')"
         />
 
+        <form-group-show-manage-devices-on-max-nodes namespace="show_manage_devices_on_max_nodes"
+          :column-label="$i18n.t('Show manage devices options when user reaches max nodes')"
+          :text="$i18n.t('When the user reaches the maximum amount of nodes for his username, enabling this will display an option in the portal to access the status page to manage/unregister devices.')"
```
Comment (extrafu) [inline html/pfappserver/root/src/views/Configuration/connectionProfiles/_components/TheForm.vue]: When the user reaches the maximum amount of nodes for his username => When the user reaches the maximum amount of registered nodes associated with his or her username

Comment (julsemaan) [reply html/pfappserver/root/src/views/Configuration/connectionProfiles/_components/TheForm.vue]: What about non binary?

File: html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm [hunk 2]
```diff
@@ -244,6 +244,11 @@ sub apply_new_node_info {
     my ($self) = @_;
     get_logger->debug(sub { use Data::Dumper; "Applying new node_info to user ".Dumper($self->new_node_info)});
 
+    if(pf::node::is_max_reg_nodes_reached($self->current_mac, $self->new_node_info->{pid}, $self->new_node_info->{category})) {
+        $self->render("max-nodes-reached.html", { show_manage_devices => $self->app->profile->showManageDevicesOnMaxNodes });
```
Comment (fdurand) [inline html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm:248]: It looks that max-nodes-reached.html is missing.

Comment (julsemaan) [reply html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm:248]: fixed

**Pull request merged**